Wb server toc byob (#625)
wandb/docs: docs/guides/hosting/secure-storage-connector.md

---
displayed_sidebar: default
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# BYOB (Secure storage connector)

> Bring your own bucket (BYOB) to store artifacts and other sensitive data for Dedicated Cloud deployments or Self-Managed instances. Data that is stored in your own bucket does not flow out of your own infrastructure.

Reviewer: ```suggestion
Bring your own bucket (BYOB) to store artifacts and other sensitive data for Dedicated Cloud deployments or Self-Managed instances. In case of Dedicated Cloud, such data is not stored in the W&B managed infrastructure.
```

Reviewer: I've changed this because `does not flow out` by itself can be slightly dodgy given changing things at our end.

> Any communication between W&B and your bucket occurs only with pre-signed URLs.

Reviewer: I think we should remove the term `only` :)

:::info

> W&B uses a garbage collection process to delete W&B Artifacts. For more information, see [How to enable garbage collection based on how W&B is hosted](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-wb-is-hosted).

Reviewer: Actually may make sense to link just to `../artifacts/delete-artifacts.md` since that also refers to TTL for scheduling deletion.

:::

## Configuration options

> There are two ways you can manage your storage object: at an Instance level or at a Team level. 

Reviewer: ```suggestion
There are two ways you can configure your storage bucket: at an Instance level or at a Team level. 
```

> - Instance level: Any user within your organization can access any runs or files stored in your W&B Server instance.

Reviewer: ```suggestion
- Instance level: Any user with relevant permissions within your organization can access any files stored in your instance-level storage.
```

> - Team level:  A storage object that only members of a specific team within your organization can access.  Team level storage objects allow greater data access control and data isolation for teams with highly sensitive data or strict compliance requirements.

Reviewer: ```suggestion
- Team level:  Storage bucket that only members of specific team(s) within your organization can access.  Team level storage buckets allow greater data access control and data isolation for teams with sensitive data or strict compliance requirements.
```

You can configure your bucket at both the instance level and separately for one or more teams within your organization.

:::tip

> Configure your instance or team level storage objects with the same cloud provider. It is not possible to configure buckets from a different cloud providers.

Reviewer: I think we can remove this tip. We're making a change to allow connecting to different cloud providers or on-prem storages. We could add that as a simple add-on in a separate PR later.

:::

> For example, suppose you have a team called Kappa in your organization. Your organization (and Team Kappa) have an instance level storage object they use. Next, you create a team called Omega. When you create Team Omega, you create a Team level storage object. Files generated by team Omega are not accessible by Team Kappa. However, files created by Team Kappa are accessible by Team Omega.

Reviewer: ```suggestion
For example, suppose you have a team called Kappa in your organization. Your organization (and Team Kappa) have an instance level storage bucket they use. Next, you create a team called Omega. When you create Team Omega, you configure a Team level storage bucket for that team. Files generated by team Omega are not accessible by Team Kappa. However, files created by Team Kappa are accessible by Team Omega. If you want data isolation for Team Kappa as well, configure a Team level storage bucket for that team as well.
```

:::tip

> Team level storage objects provides the same benefits for Self-Managed instances, especially when different business units and departments share an instance to efficiently utilize the infrastructure and administrative resources. This also applies to firms that have separate project teams managing ML workflows for separate customer engagements.

Reviewer: ```suggestion
Team level storage bucket provides the same benefits for Self-Managed instances, especially when different business units and departments share an instance to efficiently utilize the infrastructure and administrative resources. This also applies to firms that have separate project teams managing ML workflows for separate customer engagements.
```

  values={[
    {label: 'Team level', value: 'team'},
    {label: 'Instance level', value: 'instance'},
  ]}>
  <TabItem value="team">

Configure a cloud storage bucket at the Team level when you create a W&B Team:

1. Provide a name for your team in the **Team Name** field. 
2. Choose the Company or Organization you want this team to belong to from the **Company/Organization** dropdown.  
3. Select **External Storage** for the **Choose storage type** option. 
4. Choose either **New bucket** from the dropdown or select an existing bucket.
:::tip
Multiple W&B Teams can use the same cloud storage bucket. To enable this, select an existing cloud storage bucket from the dropdown.
:::
5. From the **Cloud provider** dropdown, select your cloud provider.
6. Provide the name of your storage object for the **Name** field.
7. (Optional if you use AWS) Provide the ARN of your encryption key for the **KMS key ARN** field. 
8. Select the **Create Team** button.

![](/images/hosting/prod_setup_secure_storage.png)

An error or warning appears at the bottom of the page if there are issues accessing the bucket or the bucket has invalid settings.

  </TabItem>
  <TabItem value="instance">

> Reach out to your W&B Team to configure the Instance level secure storage connector for your Dedicated Cloud or Self-managed instance.

Author: @abhinavg6 Quick question: so as to not confuse the word "Team" can we instead have their CTA be to reach out to Support?

Reviewer: Sure Noah, that works.

> |----------------------------|--------------------|----------------|------------------------|

Reviewer: Just FYI, there are going to be a few minor changes here by April, but those can come in a separate PR.

> | Dedicated Cloud | X | X | Both the instance and team level secure storage connector are available for Amazon Web Services and Google Cloud Platform. Only instance level secure storage connector is available for Azure cloud. |

Reviewer: ```suggestion
| Dedicated Cloud | X | X | Both the instance and team level BYOB are available for Amazon Web Services and Google Cloud Platform. Only instance level BYOB is available for Azure cloud. |
```

> | SaaS Cloud | | X | The team level secure storage connector is available only for Amazon Web Services and Google Cloud Platform. W&B fully manages the default and only bucket for Azure cloud. |

Reviewer: ```suggestion
| SaaS Cloud | | X | The team level secure storage connector is available only for Amazon Web Services and Google Cloud Platform. W&B fully manages the default and only storage bucket for Azure cloud. |
```

> | SaaS Cloud | | X | The team level secure storage connector is available only for Amazon Web Services and Google Cloud Platform. W&B fully manages the default and only bucket for Azure cloud. |

Reviewer: ```suggestion
| SaaS Cloud | | X | The team level BYOB is available only for Amazon Web Services and Google Cloud Platform. W&B fully manages the default and only bucket for Azure cloud. |
```
